checker/src/main/java/org/checkerframework/checker/resourceleak/MustCallConsistencyAnalyzer.java

```diff
@@ -0,0 +1,1604 @@
+    }
+    return false;
+  }
+
+  /**
+   * Returns true if this node represents a method invocation of a must-call-alias method, where the
+   * other must call alias is some ignorable pointer, such as an owning field or a pointer that is
+   * guaranteed to be non-owning, such as "`this`" or a non-owning field.
+   *
+   * @param node a method invocation node
+   * @return true if this is the invocation of a method whose return type is MCA with an owning
+   *     field or a non-owning pointer
+   */
+  private boolean returnTypeIsMustCallAliasWithIgnorable(MethodInvocationNode node) {
+    Node mcaParam = getMustCallAliasParamVar(node);
+    return mcaParam instanceof FieldAccessNode || mcaParam instanceof ThisNode;
+  }
+
+  /**
+   * Checks if {@code node} is nested inside a {@link TypeCastNode} or a {@link
+   * TernaryExpressionNode}, by looking at the successor block in the CFG.
+   *
+   * @param node the CFG node
+   * @return {@code true} if {@code node} is in a {@link SingleSuccessorBlock} {@code b}, the first
+   *     {@link Node} in {@code b}'s successor block is a {@link TypeCastNode} or a {@link
+   *     TernaryExpressionNode}, and {@code node} is an operand of the successor node; {@code false}
+   *     otherwise
+   */
+  private boolean nestedInCastOrTernary(Node node) {
+    if (!(node.getBlock() instanceof SingleSuccessorBlock)) {
+      return false;
+    }
+    Block successorBlock = ((SingleSuccessorBlock) node.getBlock()).getSuccessor();
+    if (successorBlock != null) {
+      List<Node> succNodes = successorBlock.getNodes();
+      if (succNodes.size() > 0) {
+        Node succNode = succNodes.get(0);
+        if (succNode instanceof TypeCastNode) {
+          return ((TypeCastNode) succNode).getOperand().equals(node);
+        } else if (succNode instanceof TernaryExpressionNode) {
+          TernaryExpressionNode ternaryExpressionNode = (TernaryExpressionNode) succNode;
+          return ternaryExpressionNode.getThenOperand().equals(node)
+              || ternaryExpressionNode.getElseOperand().equals(node);
+        }
+      }
+    }
+    return false;
+  }
+
+  /**
+   * Transfers ownership of locals to {@code @Owning} parameters at a method or constructor call.
+   *
+   * @param facts the current set of facts
+   * @param node a method or constructor invocation node
+   */
+  private void doOwnershipTransferToParameters(
+      Set<ImmutableSet<LocalVarWithTree>> facts, Node node) {
+
+    if (checker.hasOption(MustCallChecker.NO_LIGHTWEIGHT_OWNERSHIP)) {
+      // Never transfer ownership to parameters, matching ECJ's default.
+      return;
+    }
+
+    List<Node> actualParams = getArgumentsOfMethodOrConstructor(node);
+    List<? extends VariableElement> formalParams = getFormalsOfMethodOrConstructor(node);
+
+    if (actualParams.size() != formalParams.size()) {
+      // This could happen, e.g., with varargs, or with strange cases like generated Enum
```

Comment:
I actually don't think that this would be a problem - this code should correctly transfer ownership of the local to the `@Owning` varargs parameter, if such a thing existed, when there is only a single varargs argument. If however there are e.g. two varargs arguments, the second (and later) arguments won't get transferred - that's what this comment is about.

So for example, say you have some method with the signature `m(@Owning Socket... sockets)`. Then the call `m(mySocket)` will correctly transfer `mySocket` to `m`; but in the call `m(mySocket1, mySocket2)`, only `mySocket1` gets transferred.

I'll update this comment to draw this out better.